code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -20,7 +20,7 @@
 
 /obj/item/borg/upgrade/proc/locate_component(var/obj/item/C, var/mob/living/silicon/robot/R, var/mob/living/user)
 	if(!C || !R || !user)
-		return FAILED_TO_ADD
+		return
```

Comment:
Return `null` explicitly. A bare `return` implies that this proc doesn't return anything ever.